systemtest/edu/washington/escience/myria/systemtest/DbDeleteTest.java

```diff
@@ -0,0 +1,124 @@
+/**
+ *
+ */
+package edu.washington.escience.myria.systemtest;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.nio.file.Paths;
+import java.util.Set;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import edu.washington.escience.myria.RelationKey;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.io.DataSource;
+import edu.washington.escience.myria.io.FileSource;
+import edu.washington.escience.myria.operator.TupleSource;
+import edu.washington.escience.myria.operator.network.partition.RoundRobinPartitionFunction;
+import edu.washington.escience.myria.util.JsonAPIUtils;
+
+/**
+ * Test DbDelete Operator
+ * 
+ * This test performs a DbDelete under two different scenarios:
+ * 
+ * testDeleteRelationInCatalog(): This tests whether the Catalog deletes the relation after calling deleteDataset() from
+ * the server
+ * 
+ * testDeleteNonExistentRelation(): This tests whether deleteDataset() is able to delete the tables from the underlying
+ * database even if it does not exist in all workers expressed by the Catalog.
+ * 
+ */
+public class DbDeleteTest extends SystemTestBase {
+  /**
+   * Source of tuples
+   */
+  TupleSource relationSource;
+
+  /**
+   * The relation to ingest
+   */
+  RelationKey relationKey;
+
+  /**
+   * Schema of the ingested relation
+   */
+  Schema relationSchema;
+
+  /**
+   * Tests if the relation has been deleted from the Catalog successfully.
+   * 
+   * @throws Exception
+   */
+  @Test
+  public void testDeleteRelationInCatalog() throws Exception {
+    ingestTestDataset();
+
+    Set<Integer> workers = server.getWorkersForRelation(relationKey, null);
+    assertTrue(workers.size() == workerIDs.length);
+
+    JsonAPIUtils.deleteDataset("localhost", masterDaemonPort, relationKey.getUserName(), relationKey.getProgramName(),
+        relationKey.getRelationName());
+
+    workers = server.getWorkersForRelation(relationKey, null);
```

Comment:
Does this actually test if the relation has been deleted from the catalog entirely, or just that no partitions exist?